tree_sitter_grep_filter_before_line_number/src/lib.rs

```diff
@@ -0,0 +1,25 @@
+use std::{
+    ffi::CStr,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+use libc::c_char;
+use tree_sitter::Node;
+
+static ROW_NUMBER: AtomicUsize = AtomicUsize::new(0);
```

Comment:
Most of this stuff is only necessary to wire up the configuration argument

Eg here I went with a `static` `AtomicUsize` for (globally, which the API seems to force you to do, which seems fine?) remembering the passed configuration argument